server/src/main/java/io/unitycatalog/server/service/PermissionService.java

```diff
@@ -0,0 +1,278 @@
+import java.util.stream.Collectors;
+
+import static io.unitycatalog.server.model.SecurableType.CATALOG;
+import static io.unitycatalog.server.model.SecurableType.FUNCTION;
+import static io.unitycatalog.server.model.SecurableType.METASTORE;
+import static io.unitycatalog.server.model.SecurableType.REGISTERED_MODEL;
+import static io.unitycatalog.server.model.SecurableType.SCHEMA;
+import static io.unitycatalog.server.model.SecurableType.TABLE;
+import static io.unitycatalog.server.model.SecurableType.VOLUME;
+
+@ExceptionHandler(GlobalExceptionHandler.class)
+public class PermissionService {
+
+  private final UnityCatalogAuthorizer authorizer;
+  private static final MetastoreRepository METASTORE_REPOSITORY = MetastoreRepository.getInstance();
+  private static final UserRepository USER_REPOSITORY = UserRepository.getInstance();
+  private static final CatalogRepository CATALOG_REPOSITORY = CatalogRepository.getInstance();
+  private static final SchemaRepository SCHEMA_REPOSITORY = SchemaRepository.getInstance();
+  private static final TableRepository TABLE_REPOSITORY = TableRepository.getInstance();
+  private static final FunctionRepository FUNCTION_REPOSITORY = FunctionRepository.getInstance();
+  private static final VolumeRepository VOLUME_REPOSITORY = VolumeRepository.getInstance();
+  private static final ModelRepository MODEL_REPOSITORY = ModelRepository.getInstance();
+
+  public PermissionService(UnityCatalogAuthorizer authorizer) {
+    this.authorizer = authorizer;
+  }
+
+  // TODO: Refactor these endpoints to use a common method with dynamic resource id lookup
+  @Get("/metastore/{name}")
+  public HttpResponse getMetastoreAuthorization(
+      @Param("name") String name) {
+    return getAuthorization(METASTORE, name);
+  }
+
+  @Get("/catalog/{name}")
+  public HttpResponse getCatalogAuthorization(
+      @Param("name") String name) {
+    return getAuthorization(CATALOG, name);
+  }
+
+  @Get("/schema/{name}")
+  public HttpResponse getSchemaAuthorization(
+      @Param("name") String name) {
+    return getAuthorization(SCHEMA, name);
+  }
+
+  @Get("/table/{name}")
+  public HttpResponse getTableAuthorization(
+      @Param("name") String name) {
+    return getAuthorization(TABLE, name);
+  }
+
+  public HttpResponse getFunctionAuthorization(
+      @Param("name") String name) {
+    return getAuthorization(FUNCTION, name);
+  }
+
+  @Get("/volume/{name}")
+  public HttpResponse getVolumeAuthorization(
+      @Param("name") String name) {
+    return getAuthorization(VOLUME, name);
+  }
+
+  @Get("/registered-model/{name}")
+  public HttpResponse getRegisteredModelAuthorization(
+      @Param("name") String name) {
+    return getAuthorization(REGISTERED_MODEL, name);
+  }
+
+  private HttpResponse getAuthorization(
+      SecurableType securableType, String name) {
+
+    // Only show permissions for the authentiated identity unless they are the owner
+    // of the resource or the metastore itself.
```

Comment:
> owner of the resource or the metastore itself.

Owners of the resource's schema and catalog can also see all permissions on the resource.